baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/LoggerInterpolationConsumesThrowable.java

```diff
@@ -0,0 +1,109 @@
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.LiteralTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import java.util.List;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "LoggerInterpolationConsumesThrowable",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = SeverityLevel.ERROR,
+        summary = "Logger call interpolation markers should not be used for the throwable parameter because they "
+                + "prevent stack traces from being logged in favor of the string value of the Throwable.")
+public final class LoggerInterpolationConsumesThrowable extends BugChecker implements MethodInvocationTreeMatcher {
+
+    private static final Matcher<ExpressionTree> LOG_METHOD = MethodMatchers.instanceMethod()
+            .onDescendantOfAny("org.slf4j.Logger", "com.palantir.logsafe.logger.SafeLogger")
+            .namedAnyOf("trace", "debug", "info", "warn", "error", "fatal");
+
+    private static final Matcher<ExpressionTree> MARKER = MoreMatchers.isSubtypeOf("org.slf4j.Marker");
+    private static final Matcher<ExpressionTree> THROWABLE = MoreMatchers.isSubtypeOf(Throwable.class);
+
+    @Override
+    public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+        if (!LOG_METHOD.matches(tree, state)) {
+            return Description.NO_MATCH;
+        }
+
+        List<? extends ExpressionTree> args = tree.getArguments();
+        // No need to verify logging calls with only a message
+        if (args.size() <= 1) {
+            return Description.NO_MATCH;
+        }
+
+        // Not problematic unless slf4j curly braces consume a throwable, so we can short-circuit out.
+        int throwableIndex = args.size() - 1;
+        ExpressionTree lastArg = args.get(throwableIndex);
+        if (!THROWABLE.matches(lastArg, state)) {
+            return Description.NO_MATCH;
+        }
+
+        boolean hasMarker = MARKER.matches(tree.getArguments().get(0), state);
+        int messageIndex = hasMarker ? 1 : 0;
+        ExpressionTree messageArg = args.get(messageIndex);
+
+        if (messageArg.getKind() != Tree.Kind.STRING_LITERAL) {
+            return Description.NO_MATCH;
+        }
+
+        String literalMessage = (String) ((LiteralTree) messageArg).getValue();
+        int stringPlaceholders = countPlaceholders(literalMessage);
+        int nonThrowableParameters = throwableIndex - messageIndex - 1;
+        if (stringPlaceholders <= nonThrowableParameters) {
+            return Description.NO_MATCH;
+        }
+        int extraPlaceholders = stringPlaceholders - nonThrowableParameters;
+        return buildDescription(tree)
```

Comment:
There are a few common patterns we could likely auto-fix, but enough cases we can't that I haven't attempted to implement one.

If we find a lot of failures ending with ` {}` or `: {}` it may be reasonable to automatically update them.

Reply:
Might be interesting to track how many excavators fail to upgrade with this check, given we can't autofix them